oxbow/src/gff.rs

```diff
@@ -0,0 +1,147 @@
+use std::fs::File;
+use std::io::{BufReader, Read, Seek};
+use std::sync::Arc;
+
+use arrow::array::{ArrayRef, Float32Builder, GenericStringBuilder, Int32Builder};
+use arrow::error::ArrowError;
+use arrow::record_batch::RecordBatch;
+use noodles::gff;
+
+use crate::batch_builder::{write_ipc_err, BatchBuilder};
+
+/// A GFF reader.
+pub struct GffReader<R> {
+    reader: gff::Reader<BufReader<R>>,
+}
+
+impl GffReader<BufReader<File>> {
+    /// Creates a GFF reader from a given file path.
+    pub fn new_from_path(path: &str) -> std::io::Result<Self> {
+        let reader = File::open(path)
+            .map(BufReader::new)
+            .map(BufReader::new)
```

Comment:
I'd like to avoid this nested `BufReader` if possible, but this is what I ended up with between `new` and `new_from_path`.